examples/custom-metrics/src/main/java/com/spotify/zoltar/examples/metrics/CustomMetricsExample.java

```diff
@@ -0,0 +1,186 @@
+
+    static CustomPredictorMetrics create(final SemanticMetricRegistry registry,
+                                         final MetricId metricId) {
+      final LoadingCache<Model.Id, CustomMetrics> metersCache =
+          CacheBuilder.<Model.Id, CustomMetrics>newBuilder()
+              .build(new CacheLoader<Model.Id, CustomMetrics>() {
+                @Override
+                public CustomMetrics load(final Model.Id id) {
+                  return CustomMetrics.create(registry, metricId.tagged("model", id.value()));
+                }
+              });
+
+      return new AutoValue_CustomMetricsExample_CustomPredictorMetrics(metersCache);
+    }
+
+    @Override
+    public PredictFnMetrics<Integer, Float> predictFnMetrics() {
+      return id -> {
+        final CustomMetrics metrics = metricsCache().getUnchecked(id);
+        final Counter negativePredictCounter = metrics.negativePredictCount();
+        return NegativePredictMetrics.create(negativePredictCounter);
+      };
+    }
+
+    @Override
+    public FeatureExtractorMetrics<Integer, Float> featureExtractorMetrics() {
+      return id -> {
+        final CustomMetrics metrics = metricsCache().getUnchecked(id);
+        final Counter negativeExtractCounter = metrics.negativeExtractCount();
+        return NegativeExtractMetrics.create(negativeExtractCounter);
+      };
+    }
+  }
+
+  /**
+   * To define a metric you want to measure for prediction, implement PredictMetrics.
+   */
+  @AutoValue
+  abstract static class NegativePredictMetrics implements PredictMetrics<Integer, Float> {
+    abstract Counter negativePredictCount();
+
+    static NegativePredictMetrics create(final Counter negativeCount) {
+      return new AutoValue_CustomMetricsExample_NegativePredictMetrics(negativeCount);
+    }
+
+    /**
+     * Here, given a list of predictions, we want to measure the number of negatives.
+     */
+    @Override
+    public void prediction(final List<Prediction<Integer, Float>> predictions) {
+      negativePredictCount().inc(predictions.stream().filter(x -> x.value() < 0).count());
+    }
+  }
+
+  /**
+   * To define a metric you want to measure for feature extraction, implement VectorMetrics.
+   */
+  @AutoValue
+  abstract static class NegativeExtractMetrics implements VectorMetrics<Integer, Float> {
+    abstract Counter negativeExtractCount();
+
+    static NegativeExtractMetrics create(final Counter negativeCount) {
+      return new AutoValue_CustomMetricsExample_NegativeExtractMetrics(negativeCount);
+    }
+
+    /**
+     * Here, given a list of feature extraction results, we want to count the number of negatives.
+     */
+    @Override
+    public void extraction(final List<Vector<Integer, Float>> vectors) {
+      negativeExtractCount().inc(vectors.stream().filter(v -> v.value() < 0).count());
+    }
+  }
+
+  CustomMetricsExample(final SemanticMetricRegistry metricRegistry, final MetricId metricId)
```

Comment:
@andrewsmartin wdyt of making this class runnable? or we can make `CustomMetricsExample` implement `Predictor` ... that way would be a predictor with built in metrics? than you can remove the custom predict at L182. wdyt?